Alc/ALc.c

```diff
@@ -4669,20 +4669,33 @@ ALC_API const ALCchar* ALC_APIENTRY alcGetStringiSOFT(ALCdevice *device, ALCenum
 ALC_API ALCboolean ALC_APIENTRY alcResetDeviceSOFT(ALCdevice *device, const ALCint *attribs)
 {
     ALCenum err;
+    enum DeviceConnect Connected;
 
     LockLists();
-    if(!VerifyDevice(&device) || device->Type == Capture ||
-       ATOMIC_LOAD(&device->Connected, almemory_order_relaxed) != DeviceConnect_Connected)
+    if(!VerifyDevice(&device) || device->Type == Capture)
     {
         UnlockLists();
         alcSetError(device, ALC_INVALID_DEVICE);
         if(device) ALCdevice_DecRef(device);
         return ALC_FALSE;
     }
     almtx_lock(&device->BackendLock);
+    Connected = ATOMIC_LOAD(&device->Connected, almemory_order_relaxed);
     UnlockLists();
+    if (Connected == DeviceConnect_Disconnected)
+    {
+        // try reconnect..
+        Connected = DeviceConnect_Connecting;
+        ATOMIC_STORE(&device->Connected, DeviceConnect_Connecting, almemory_order_relaxed);
+    }
 
     err = UpdateDeviceParams(device, attribs);
+
+    if (err == ALC_NO_ERROR)
+    {
+        ATOMIC_COMPARE_EXCHANGE_WEAK(&device->Connected, &Connected, DeviceConnect_Connected, almemory_order_acq_rel, almemory_order_acquire);
```

Comment:
This doesn't look quite right. This almost certainly should be `_STRONG` (weak cmpxchg is allowed to spuriously fail and not swap, even if the comparison is true). There's also a bit of a race condition since you changed the backends' mixing loops to check for `DeviceConnect_Connected`, so if the backend loop happens to run and see `DeviceConnect_Connecting` before this gets a chance to change it, it'll stop mixing despite having successfully restarted.